site-permissions/site-permissions-impl/src/main/java/com/duckduckgo/site/permissions/impl/SitePermissionsDialogActivityLauncher.kt

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.site.permissions.impl
+
+import android.Manifest
+import android.app.AlertDialog
+import android.content.Context
+import android.net.Uri
+import android.view.LayoutInflater
+import android.view.View
+import android.webkit.PermissionRequest
+import android.widget.TextView
+import android.widget.Toast
+import androidx.activity.result.ActivityResultCaller
+import androidx.annotation.StringRes
+import com.duckduckgo.di.scopes.ActivityScope
+import com.duckduckgo.site.permissions.api.SitePermissionsDialogLauncher
+import com.duckduckgo.site.permissions.impl.R.layout
+import com.squareup.anvil.annotations.ContributesBinding
+import dagger.SingleInstanceIn
+import javax.inject.Inject
+
+@ContributesBinding(ActivityScope::class)
+@SingleInstanceIn(ActivityScope::class)
```

Comment:
Does this need to be a singleton? If my comment below is true and the object doesn't have a state I don't think it needs to be a singleton.

Reply:
This is a Launcher Dialog class which will need the activity and also persist permissionRequest and some other parameters for setting app a couple of callbacks for when systemPermissions are allowed or denied.
Because those callbacks are registered in onCreate(), then we need to set those values when a site permission is requested so the callback can access them.
I think this is the cleaner and the easiest way to implement it, but I'm open to suggestions :)

Reply:
It's still not clear to me why if we want to share 1 instance between all fragments, and what's the difference if we have 1 instance per fragment. Let's discuss about it offline.